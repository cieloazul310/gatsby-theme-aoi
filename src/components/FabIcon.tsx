import * as React from 'react';
import Icon, { IconProps } from '@mui/material/Icon';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface Props extends IconProps {
  icon: IconDefinition;
  defaultVertical?: boolean;
}

// helpers
function sizeToFontSize(fontSize: IconProps['fontSize']): number {
  if (fontSize === 'small') return 20;
  if (fontSize === 'large') return 36;
  return 24;
}

function FabIcon({ icon, defaultVertical, ...options }: Props): JSX.Element {
  const [width] = icon.icon;
  const dy = width > 512 ? (512 - width) / 2 : 0;
  return (
    <Icon {...options}>
      <FontAwesomeIcon
        icon={icon}
        style={{
          verticalAlign: !defaultVertical ? '0.125em' : undefined,
          transform: dy !== null ? `translateX(${dy / sizeToFontSize(options.fontSize)}px)` : undefined,
        }}
      />
    </Icon>
  );
}

FabIcon.defaultProps = {
  defaultVertical: undefined,
};

export default FabIcon;
