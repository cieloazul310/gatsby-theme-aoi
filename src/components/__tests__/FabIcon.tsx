import * as React from 'react';
import renderer from 'react-test-renderer';
import { faTwitter, faYoutubeSquare } from '@fortawesome/free-brands-svg-icons';
import FabIcon from '../FabIcon';

describe('FabIcon', () => {
  it('TwitterIcon', () => {
    const tree = renderer.create(<FabIcon icon={faTwitter} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Large Icon', () => {
    const tree = renderer.create(<FabIcon icon={faTwitter} fontSize="large" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('default vertical', () => {
    const tree = renderer.create(<FabIcon icon={faTwitter} defaultVertical={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('disable default vertical', () => {
    const tree = renderer.create(<FabIcon icon={faTwitter} defaultVertical={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Wide Icon', () => {
    const tree = renderer.create(<FabIcon icon={faYoutubeSquare} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
