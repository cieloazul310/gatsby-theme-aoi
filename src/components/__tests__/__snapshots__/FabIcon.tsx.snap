// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FabIcon Large Icon 1`] = `
<span
  aria-hidden={true}
  className="material-icons MuiIcon-root MuiIcon-fontSizeLarge"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-twitter fa-w-16 "
    data-icon="twitter"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={
      Object {
        "transform": "translateX(0px)",
        "verticalAlign": "0.125em",
      }
    }
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`FabIcon TwitterIcon 1`] = `
<span
  aria-hidden={true}
  className="material-icons MuiIcon-root"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-twitter fa-w-16 "
    data-icon="twitter"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={
      Object {
        "transform": "translateX(0px)",
        "verticalAlign": "0.125em",
      }
    }
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`FabIcon Wide Icon 1`] = `
<span
  aria-hidden={true}
  className="material-icons MuiIcon-root"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-youtube-square fa-w-14 "
    data-icon="youtube-square"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={
      Object {
        "transform": "translateX(0px)",
        "verticalAlign": "0.125em",
      }
    }
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M186.8 202.1l95.2 54.1-95.2 54.1V202.1zM448 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zm-42 176.3s0-59.6-7.6-88.2c-4.2-15.8-16.5-28.2-32.2-32.4C337.9 128 224 128 224 128s-113.9 0-142.2 7.7c-15.7 4.2-28 16.6-32.2 32.4-7.6 28.5-7.6 88.2-7.6 88.2s0 59.6 7.6 88.2c4.2 15.8 16.5 27.7 32.2 31.9C110.1 384 224 384 224 384s113.9 0 142.2-7.7c15.7-4.2 28-16.1 32.2-31.9 7.6-28.5 7.6-88.1 7.6-88.1z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`FabIcon default vertical 1`] = `
<span
  aria-hidden={true}
  className="material-icons MuiIcon-root"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-twitter fa-w-16 "
    data-icon="twitter"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={
      Object {
        "transform": "translateX(0px)",
        "verticalAlign": undefined,
      }
    }
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`FabIcon disable default vertical 1`] = `
<span
  aria-hidden={true}
  className="material-icons MuiIcon-root"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-twitter fa-w-16 "
    data-icon="twitter"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={
      Object {
        "transform": "translateX(0px)",
        "verticalAlign": "0.125em",
      }
    }
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;
